### ISSUE TITLE
Imagenet example: cutorch seed not set

### ISSUE THREAD

**Issue by mys007** [facebook/fbcunn#55]
`fbcunn/examples/imagenet/main.lua` doesn't set the pseudorandom seed for CUDA, something like `cutorch.manualSeed(opt.manualSeed)`. This leads to unrepeatable experiments. With the fix, repeatable experiments seem to be possible at least with the ccn2 back-end, according to my experience.

**Comment by soumith** [comment]
hmm, I didn't think that I was even using cuda rng anywhere in the code.

**Comment by soumith** [comment]
cudnn convolutions are not deterministic (as confirmed by nvidia), maxpooling as well (but that's under debate in the other thread and I haven't asked nvidia). So probably with the cudnn backend it's not repeatable. ccn2 is indeed deterministic I guess.

**Comment by mys007** [comment]
I think that at least the modules might be influenced by cuda rng, nn.Dropout in particular, isn't it the case? Of course, I'm talking about the case when not using donkeys, as the order of their executions is non-deterministic by principle due to no synchronization primitives.

**Comment by soumith** [comment]
that's true, I forgot about nn.Dropout
